engine/src/test/java/io/camunda/zeebe/engine/processing/incident/ServiceTaskIncidentTest.java

```diff
@@ -286,4 +286,49 @@ public void shouldResolveIncidentAfterJobRetriesExpressionEvaluationFailed() {
 
     assertThat(incidentResolvedEvent.getKey()).isEqualTo(incidentRecord.getKey());
   }
+
+  @Test
+  public void shouldResolveIncidentAndCreateNewIncidentWhenContinuationFails() {
+    // given a deployed process with a service task with an output expression
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess("process-incident-on-leaving-task")
+                .startEvent()
+                .serviceTask(
+                    "task",
+                    b -> b.zeebeJobType("task").zeebeInputExpression("unknown_var", "output"))
+                .done())
+        .deploy();
+
+    // and an instance of that process is created without a variable for the output expression
+    final var processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId("process-incident-on-leaving-task").create();
+
+    // and an incident created
+    final var incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    // when we try to resolve the incident
+    final var resolveResponse =
+        ENGINE.incident().ofInstance(processInstanceKey).withKey(incident.getKey()).resolve();
+
+    assertThat(
+            RecordingExporter.incidentRecords(IncidentIntent.RESOLVED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withRecordKey(incident.getKey())
+                .exists())
+        .describedAs("then the incident is resolved")
```

Comment:
These methods are more to assign labels to objects, not to narrate a story. I think the error message will sound weird if it fails. Something like "the condition 'then the incident is resolved' was not met; expected: true, actual: false"

Could you please test how the failure message looks like and whether it makes sense?
Maybe, I am being too picky here.

Maybe something like `describedAs("original incident is resolved")` would work better, same for next message.

Reply:
To check, I've flipped the assertion (so isFalse):
```
org.opentest4j.AssertionFailedError: [then the incident is resolved] 
Expecting value to be false but was true
Expected :false
Actual   :true
```

So, failure would look like:
```
org.opentest4j.AssertionFailedError: [then the incident is resolved] 
Expecting value to be true but was false
Expected :true
Actual   :false
```

With your suggestion that would look like:
```
org.opentest4j.AssertionFailedError: [original incident is resolved] 
Expecting value to be true but was false
Expected :true
Actual   :false
```

I'm not really for or against this change. Both is fine by me. Let me know what you prefer.

Reply:
I'd be open to better assertions, where it's not just comparing true and false. I'm unsure what that should look like

Reply:
Works for me the way it is.